InscryptionAPI/Card/CardExtensions.cs

```diff
@@ -396,275 +583,429 @@ public static CardInfo SetTail(this CardInfo info, string tailName, Texture2D ta
         return info;
     }
 
+    #endregion
+
+    #region CardCosts
+
     /// <summary>
-    /// Sets the ice cube parameters of the card. These parameters are used to make the IceCube ability function correctly.
+    /// Sets the cost of the card. Any and all costs can be set this way.
     /// </summary>
-    /// <param name="iceCube">The card that will be generated when this card dies.</param>
-    /// <param name="mods">A set of card mods to be applied to the ice cube contents</param>
-    /// <returns></returns>
-    public static CardInfo SetIceCube(this CardInfo info, CardInfo iceCube, IEnumerable<CardModificationInfo> mods = null)
+    /// <param name="info">Card to access</param>
+    /// <param name="bloodCost">The cost in blood (sacrifices)</param>
+    /// <param name="bonesCost">The cost in bones</param>
+    /// <param name="energyCost">The cost in energy</param>
+    /// <param name="gemsCost">The cost in gems</param>
+    /// <returns>The same card info so a chain can continue</returns>
+    public static CardInfo SetCosts(this CardInfo info, int? bloodCost = 0, int? bonesCost = 0, int? energyCost = 0, List<GemType> gemsCost = null)
```

Comment:
This is technically a better name, but it is a breaking change. I think it should stay SetCost just because the value of getting the name 100% probably doesn't outweigh breaking things.